src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1045,6 +1045,7 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
         foreach (array('doctrine', 'psr6', 'redis') as $name) {
             if (isset($config[$name = 'default_'.$name.'_provider'])) {
                 $container->setAlias('cache.'.$name, Compiler\CachePoolPass::getServiceProvider($container, $config[$name]));
+                $container->getAlias('cache.'.$name)->setPublic(false);
```

Comment:
you could even set it as private directly instead of getting the alias again:

``setAlias('cache.'.$name, new Alias(..., false));``